Source/WebKit/UIProcess/glib/ScreenManager.h

```diff
@@ -43,20 +51,24 @@ class ScreenManager {
 public:
     static ScreenManager& singleton();
 
-    PlatformDisplayID displayID(GdkMonitor*) const;
-    GdkMonitor* monitor(PlatformDisplayID) const;
+    PlatformDisplayID displayID(PlatformMonitor*) const;
+    PlatformMonitor* monitor(PlatformDisplayID) const;
 
     WebCore::ScreenProperties collectScreenProperties() const;
 
 private:
     ScreenManager();
 
-    void addMonitor(GdkMonitor*);
-    void removeMonitor(GdkMonitor*);
+    static PlatformDisplayID generatePlatformDisplayID(PlatformMonitor*);
+
+    void addMonitor(PlatformMonitor*);
+    void removeMonitor(PlatformMonitor*);
     void propertiesDidChange() const;
 
-    Vector<GRefPtr<GdkMonitor>, 1> m_monitors;
-    HashMap<GdkMonitor*, PlatformDisplayID> m_monitorToDisplayIDMap;
+    Vector<GRefPtr<PlatformMonitor>, 1> m_monitors;
+    HashMap<PlatformMonitor*, PlatformDisplayID> m_monitorToDisplayIDMap;
```

Comment:
I see your point, maybe we can just do it now.

Reply:
We need the map, because in the case of GTK, the id is generated and not returned by the monitor, so this map is the only reference we have between a GdkMonitor and its generated id. So, we could do it for WPE, but better keep this and share the code, I think

Reply:
Ah, I see. Let's then keep it and share the code.